### DISCUSSION TITLE
Seeking Guidance: Handling Commands Ending with ';' in AsyncSSH SSH Server

### DISCUSSION THREAD

**Discussion by sugam-warrior** [ronf/asyncssh#619] [General]
Hello Everyone,
I've developed an SSH server using asyncssh, and I'm facing an issue where the client is passing commands that end with ';' instead of '\n' to the session. I'm seeking guidance on how to accommodate this behavior. Specifically, I'm wondering if there's a way to update '_keylist' of class 'SSHLineEditor' to include ';' as one of the '_end_line' characters.
Any assistance or insights on how to handle this situation would be highly appreciated!

Thank you in advance for your help.

**Comment by ronf** [comment]
The "line editor" is generally only used when the client is an interactive end-user, where you want to echo their input as they type it and provide line editing capabilities where the SSH server only sees complete lines as actual input (after optional editing).

When you talk about passing commands from the client, is it writing these commands to stdin of the SSH server session, or is it passing commands via the `command` argument? If it is using the `command` argument, there's really no need for any kind of an end-of-line character. You can end in whatever you like, as you'll always be getting the command as a whole in the new session request.

If you want to read data from stdin and treat ";" similar to a newline, you can use `readuntil()` where you pass an a delimiter of `[';', '\n']`. However, you'll probably need to disable the line editing in this case to make things work properly so the line editor doesn't end up buffering your input.
